src/utils/transactionUtils/transactionUtils.ts

```diff
@@ -326,10 +330,80 @@ export const FieldToInteractionMap: {
 } = {
   name: ANT_INTERACTION_TYPES.SET_NAME,
   ticker: ANT_INTERACTION_TYPES.SET_TICKER,
+  targetID: ANT_INTERACTION_TYPES.SET_TARGET_ID,
   // TODO: add other interactions
 };
 
 export function getInteractionTypeFromField(field: string) {
   // TODO: add contract specification and more interaction fields
   return FieldToInteractionMap[field];
 }
+
+export function mapTransactionDataKeyToPayload(
+  contractType: ContractType,
+  interactionType: AntInteraction | RegistryInteraction,
+  data: string[],
+): TransactionData | undefined {
+  const payload: any = {};
+
+  if (isRegistryInteraction(interactionType)) {
+    if (
+      !isInteractionCompatible({
+        contractType,
+        interactionType,
+        functionName:
+          TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][interactionType]
+            .functionName,
+      })
+    ) {
+      throw new Error(`Interaction is not compatible`);
+    }
+    payload.functionName =
+      TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][
+        interactionType
+      ].functionName;
+    TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][
+      interactionType
+    ].keys.forEach((key, index) => {
+      if (!data[index]) {
+        // if missing transaction data from the url, throw an error
+        throw new Error(
+          `Missing key (${key}) from transaction data in the url. This may be due to the order of the data, the current order is [${data}], the correct order is [${
+            TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][interactionType].keys
+          }]`,
+        );
+      }
+      payload[key] = data[index];
+    });
+    return payload;
```

Comment:
you _could_ use a reducer here to do this and just return an object instead of appending `payload` object. but fine for now.